config/initializers/monkeypatches/active_record_connection_handling.rb

```diff
@@ -15,26 +15,28 @@ def connection
     end
 
     def check_calling_connection_allowed
-      case ActiveRecord.permanent_connection_checkout
-      when :deprecated
-        # :nocov:
-        ActiveRecord.deprecator.warn(<<~MESSAGE)
-          Called deprecated `ActiveRecord::Base.connection` method.
+      unless $PROGRAM_NAME.match?(/\bdatabase_consistency\b/) || caller.any?(/\bimmigrant\b/)
```

Comment:
The better thing to do would be to submit PRs to these two gems (`database_consistency` and `immigrant`) to have them stop using `ActiveRecord::Base.connection`, but I don't feel like doing that.